### ISSUE TITLE
rebooting of wemosD1 mini

### ISSUE THREAD

**Issue by kpcz** [pinkywafer/PinkyLEDs#13]
Hi. first of all thank you for this great sketch. But I got a problem. My wemos D1 mini keeps restarting, if I choose effects like USA, Go blue, Christmas and other. Other effects do work good. Tried various power bricks, That does not help. Tried different D1 mini, same behavior. 

It also reboots after few minutes in every effect  :(

How can i debug the issue, or what could be the problem? Thanks

Have only 20 leds running.

got this on debug, when reboot occured:

```
MQTT message received {"state": "ON"}
MQTT message received {"state": "ON", "effect": "USA"}

Exception (9):
epc1=0x40202a12 epc2=0x00000000 epc3=0x00000000 excvaddr=0x3ffe8d5e depc=0x00000000

>>>stack>>>

ctx: cont
sp: 3ffffd40 end: 3fffffc0 offset: 01a0
3ffffee0:  0126c417 00000000 3ffef05c 40204a35  
3ffffef0:  3fffff30 00000000 3ffef5e8 3ffef830  
3fffff00:  3fffdad0 00000000 00000002 3ffef830  
3fffff10:  00000000 00000000 3ffef57c 40205087  
3fffff20:  00000000 3ffef67c 3fff0bb4 40209912  
3fffff30:  93960000 4872001f 00696a00 00008919  
3fffff40:  66001774 0759000c 04034d01 32090340  
3fffff50:  09410303 01184b02 2c003748 cc000077  
3fffff60:  a180c05b 0000e7ff 3ffef5e8 4020316a  
3fffff70:  00000096 feefeffe 000000a1 00000067  
3fffff80:  ffcd0000 22b2dcf8 feefef22 feefeffe  
3fffff90:  00000000 00000000 00000001 3ffef830  
3fffffa0:  3fffdad0 00000000 3ffef800 4020fb74  
3fffffb0:  feefeffe feefeffe 3ffe8590 40100cf5  
<<<stack<<<

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v951aeffa
~ld
⸮⸮
```

**Comment by pinkywafer** [comment]
I think this is related to #4
Perhaps the same issue. I have a feeling that either there's a memory overflow somewhere,  or something is causing blocking. I haven't had chance to further debug yet, but will get to it when I can.   I have however managed to replicate the issue, (albeit with no leds physically attached,  but programmed - I think it shouldn't matter if they actually attached or not, as the output is being bitbanged out)

**Comment by kpcz** [comment]
Strangest thing is that, like 2 month ago, I flashed wemos d1 mini, with Your sketch, and had the same problem. Then, I commented out "#define FASTLED_ALLOW_INTERRUPTS 0" line, and afterwards it was working good for two wemos D1 devices. Now I had to make some changes in mqtt topics, so I re-flashed the sketch to one of them, and it started to reboot after every few mins. Strange.

**Comment by kpcz** [comment]
https://github.com/Snipercaine/Holiday-LED-files/pull/25/commits/62eae79768a4398086b5d863bc73b196037c5ea2
this fix allows the lights to be more stable, but still not 100%. At some point it disconnects from Home Assistant. 

`#define FL_ALIGN_PROGMEM  __attribute__ ((aligned (4)))`

**Comment by pinkywafer** [comment]
Thanks for this. I've currently got a D1 running (no LEDs physically connected) but it believes it has 10,000 connected - powered only via usb... monitoring!

**Comment by kpcz** [comment]
After adding mentioned line, I haven't seen any strange behavior of my Led strip, so I will close the issue.  Thank You again for this great sketch.

**Comment by pinkywafer** [comment]
Great news. Thanks

**Comment by pinkywafer** [comment]
I'm reopening this, until I've added this to the code, as the issue still exists within the code here

**Comment by bbacskay** [comment]
This issue should be fixed in the FastLED library. See this:
FastLED/FastLED#696
Somebody tried to address it, but only for ESP32.

**Comment by pinkywafer** [comment]
@bbacskay thanks - it looks like this fix was included for esp8266 also at FastLED v 3.3.2
https://github.com/FastLED/FastLED/issues/804

I've just updated again!
